File: facturalite-web/src/types/factura.ts
// Cabecera + líneas (alineado con lo que devuelve tu API .NET)
export type FacturaLinea = {
  id: number;
  facturaId: number;
  productoId: number;
  cantidad: number;
  precioUnitario?: number;
  ivaPorcentaje?: number;
  producto?: { id: number; nombre: string; sku?: string; precioUnitario?: number; ivaPorcentaje?: number };
};

export type Factura = {
  id: number;
  numero: string;
  fecha: string;         // ISO
  clienteId: number;
  cliente?: {
    id: number;
    nombre: string;
    email?: string;
    nif?: string;
    telefono?: string;
    direccion?: string;
    fechaAlta: string;
    activo: boolean;
  };
  totalBase: number;
  totalIva: number;
  total: number;
  activa: boolean;
  lineas: FacturaLinea[];
};

File: FacturaLite.Api/Domain/Entities/Factura.cs
namespace FacturaLite.Api.Domain.Entities
{
    // Cabecera de la factura
    public class Factura
    {
        public int Id { get; set; }
        public string Numero { get; set; } = string.Empty;
        public DateTime Fecha { get; set; } = DateTime.UtcNow;

        // Relación con cliente
        public int ClienteId { get; set; }
        public Cliente? Cliente { get; set; }

        // Totales
        public decimal TotalBase { get; set; }
        public decimal TotalIva { get; set; }
        public decimal Total { get; set; }

        public bool Activa { get; set; } = true;

        // Navegación a líneas
        public List<FacturaLinea> Lineas { get; set; } = new();
    }
    
}
